src/scanpy/preprocessing/_pca/_dask.py

```diff
@@ -131,8 +131,8 @@ def transform_block(
             mean_=self.mean_,
             components_=self.components_,
             chunks=(x.chunks[0], self.n_components_),
-            meta=np.zeros([0], dtype=x.dtype),
-            dtype=x.dtype,
+            meta=np.zeros([]),
+            dtype=np.result_type(x.dtype, self.components_.dtype, self.mean_.dtype),
```

Comment:
> **dtype**: The `dtype` of the output array. […]
> **meta**: The `meta` of the output array, when specified is expected to be an array of the same type and dtype of that returned when calling `.compute()` on the array returned by this function. […]

so shouldn’t this be the following?

I don’t get why dask wants both to avoid running the function on test data. Shouldn’t `meta` be enough, and allow dask to infer `dtype` from `meta`?

```suggestion
            meta=np.zeros((), dtype=dtype_out := np.result_type(x.dtype, self.components_.dtype, self.mean_.dtype)),
            dtype=dtype_out,
```

Reply:
Yes! we can just use `meta` all good